File: apps/desktop/src-tauri/src/commands/import.rs
use tauri::State;

use crate::{
    errors::{CommandResult, IntoCommandResult},
    models::asset::{
        ImportPreviewResult, ImportRemoteImagesRequest, ImportRequest,
        ImportResult,
    },
    services::AssetService,
};

#[tauri::command]
pub async fn preview_import_images(
    payload: ImportRequest,
    asset_service: State<'_, AssetService>,
) -> CommandResult<ImportPreviewResult> {
    asset_service.preview_import_images(payload).await.into_command()
}

#[tauri::command]
pub async fn import_images(
    payload: ImportRequest,
    asset_service: State<'_, AssetService>,
) -> CommandResult<ImportResult> {
    asset_service.import_images(payload).await.into_command()
}

#[tauri::command]
pub async fn import_remote_images(
    payload: ImportRemoteImagesRequest,
    asset_service: State<'_, AssetService>,
) -> CommandResult<ImportResult> {
    asset_service.import_remote_images(payload).await.into_command()
}

File: apps/desktop/src/shared/lib/ipc/import.ts
import type {
  ImportPreviewResult,
  ImportRemoteImagesRequest,
  ImportRequest,
  ImportResult,
} from '../../types';
import { invokeCamel } from './core';

export const importIpc = {
  previewImages: (payload: ImportRequest): Promise<ImportPreviewResult> =>
    invokeCamel('preview_import_images', { payload }),
  importImages: (payload: ImportRequest): Promise<ImportResult> =>
    invokeCamel('import_images', { payload }),
  importRemoteImages: (payload: ImportRemoteImagesRequest): Promise<ImportResult> =>
    invokeCamel('import_remote_images', { payload }),
};
